contributing.md

```diff
@@ -36,74 +36,43 @@ If the code which you are working on does not yet have any documentation, we wou
 
 ### Formatting
 
-GPflow uses [black](https://github.com/psf/black) for formatting in the following way:
+GPflow uses [black](https://github.com/psf/black) for formatting, calling it with `black -t py36 -l 100`. Simply run `make format` from the GPflow root directory.
 
-```bash
-black -t py36 -l 100 gpflow tests doc setup.py
-```
+## Pull requests
 
-## Pull requests and the master branch
+If you think that your contribution falls within the project scope (see above) please submit a Pull Request (PR) to our GitHub page.
+(GitHub provides extensive documentation on [forking](https://help.github.com/en/github/getting-started-with-github/fork-a-repo) and [pull requests](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests).)
 
-If you think that your contribution falls within the project scope (see above) please submit a Pull Request (PR) to our GitHub page. In order to maintain code quality, and make life easy for the reviewers, please ensure that your PR:
+In order to maintain code quality, and make life easy for the reviewers, please ensure that your PR:
 
 - Only fixes one issue or adds one feature.
 - Makes the minimal amount of changes to the existing codebase.
+- Is testing its changes.
+- Passes all checks (formatting, types, tests - you can run them all locally using `make check-all` from the GPflow root directory).
 
 All code goes through a PR; there are no direct commits to the master and develop branches.
 
-
 ## Tests and continuous integration
 
 GPflow is ~97% covered by the testing suite. We expect changes to code to pass these tests, and for new code to be covered by new tests. Currently, tests are run by CircleCI and coverage is reported by codecov. Pull requests should aim to have >97% *patch* coverage (i.e., all the lines that are *changing* should be covered by tests).
 
 ## Documentation
 
-GPflow's documentation is not comprehensive, but covers enough to get users started. We expect that new features have documentation that can help others get up to speed. The docs are mostly IPython notebooks that compile into HTML via Sphinx, using nbsphinx.
+GPflow's documentation is not comprehensive, but covers enough to get users started. We expect that new features have documentation that can help others get up to speed. The docs are mostly IPython notebooks (stored in the git repository in jupytext format) that compile into HTML via Sphinx, using nbsphinx.
 
 ## Version numbering
 
 The main purpose of versioning GPflow is user convenience.
 
 We use the [semantic versioning scheme](https://semver.org/). The semver implies `MAJOR.MINOR.PATCH` version scheme, where `MAJOR` changes when there are incompatibilities in API, `MINOR` means adding functionality without breaking existing API and `PATCH` presumes the code update has backward compatible bug fixes.
```

Comment:
I don't think we do (and I don't think we should)...

Reply:
support semver?

Reply:
What I mean is, it's still code for research, we may find things we want to change that won't exactly be backwards compatible - if we strictly increase the major version, we'll quickly be at gpflow 10.3. Or keep lots of cruft around to maintain backwards compatibility. I'd rather go with something like Django - in general, aim to add a deprecation warning e.g. in GPflow 2.1, and then remove the feature in the following minor version, and occasionally to just say "backwards incompatibilities when upgrading:" in the release notes when it's not worth the effort to support both old and new way.

Reply:
ok. fair enough. i'm happy with either. definitely sth to ask the others about

Reply:
i'm going to resolve this - can put a ticket in the new board: "Do we want to support semver?"